src/course_bot/csa.clj

```diff
@@ -1,110 +1,114 @@
-                      "essay3-reviews" (essay/review-score conf "essay3"))
+                     (report/report-talk @db @conf
+                                         (report/report-talk @db @conf
+                                                             "ID" report/stud-id
+                                                             "name" report/stud-name
+                                                             "group" report/stud-group
+                                                             "lab1-group" (pres/report-presentation-group "lab1")
+                                                             "lab1-rank" (pres/report-presentation-avg-rank "lab1")
+                                                             "lab1-score" (pres/report-presentation-score @conf "lab1")
+                                                             "lab1-count" (pres/lesson-count "lab1")
+                                                             "failed-tests" (quiz/fail-tests @conf)
+                                                             "success-test-percent" (quiz/success-tests-percent @conf)
+                                                             "essay1" (essay/essay-score "essay1")
+                                                             "essay1-reviews" (essay/review-score @conf "essay1")
+                                                             "essay2" (essay/essay-score "essay2")
+                                                             "essay2-reviews" (essay/review-score @conf "essay2")
+                                                             "essay3" (essay/essay-score "essay3")
+                                                             "essay3-reviews" (essay/review-score @conf "essay3"))
 
-  (handlers/command "help" {{id :id} :chat} (talk/send-text (-> conf :token) id (talk/helps)))
+                                         (handlers/command "help" {{id :id} :chat} (talk/send-text (-> @conf :token) id (talk/helps)))
 
-  (handlers/message {{id :id} :chat :as message}
-                    (let [err (format (tr :csa/unknown-1) message)]
-                      (println err)
-                      (talk/send-text token id err))))
+                                         (handlers/message {{id :id} :chat :as message}
+                                                           (let [err (format (tr :csa/unknown-1) message)]
+                                                             (println err)
+                                                             (talk/send-text @token id err))))
 
-(defn run [& _args]
-  (println (tr :csa/start))
-  (loop [channel (polling/start token bot-api)]
-    (Thread/sleep 500)
-    (print (tr :csa/dot)) (flush)
-    (if (.closed? channel)
-      (do (print (tr :csa/stop))
-          (recur (polling/start token bot-api)))
-      (recur channel)))
-  (println (tr :csa/stop)))
+                     (defn -main [& _args]
```

Comment:
It actually seems to run with `defn -main`. I think that it's obligatory to specify the entry point of application like that because it will fail otherwise
`➜  course-bot git:(master) ✗ java -jar course-bot.jar
Exception in thread "main" java.lang.UnsupportedOperationException: course-bot.csa/-main not defined
`